lib/src/main/kotlin/io/provenance/entity/direct/DirectKeyEntity.kt

```diff
@@ -0,0 +1,20 @@
+package io.provenance.entity.direct
+
+import io.provenance.entity.KeyEntity
+import io.provenance.entity.KeyType
+import io.provenance.hdwallet.common.hashing.sha256
+import io.provenance.hdwallet.ec.extensions.toECPrivateKey
+import io.provenance.hdwallet.signer.BCECSigner
+import io.provenance.scope.encryption.model.DirectKeyRef
+import java.security.PrivateKey
+
+class DirectKeyEntity(
+    keyRefs: Map<KeyType, DirectKeyRef>,
+    private val privateKey: PrivateKey,
+) : KeyEntity(keyRefs) {
+    
+    override fun sign(keyType: KeyType, data: ByteArray): ByteArray =
+        BCECSigner().sign(privateKey.toECPrivateKey(), data.sha256())
```

Comment:
Spitballing here, do you need to add that snippet
```kotlin
import org.bouncycastle.jce.provider.BouncyCastleProvider
import java.security.Security

Security.addProvider(BouncyCastleProvider())
```
anywhere for this to work?

Reply:
good thought. I'm able to run this locally without needing to include the provider. `p8e-cee-api` also doesn't appear to do this, so it appears the HD wallet takes care of any setup work needed.